lib/server_connection.ml

```diff
@@ -186,6 +186,7 @@ let set_error_and_handle ?request t error =
     t.error_handler ?request error (fun headers ->
       Writer.write_response writer (Response.create ~headers status);
       Body.of_faraday (Writer.faraday writer));
+    shutdown_writer t;
```

Comment:
This will break streaming responses in that all write calls to the body will raise after `shutdown_writer` is called. It will also break asynchronous responses like in #18 since in that case the `Writer.write_response` call in callback passed to `error_handler` will be performed after `shutdown_writer` is called.

As separate test should be added for this case in a separate PR before this one moves forward. If the change itself is unnecessary, then it should also be removed.

Reply:
Do you think replacing it with `wakeup_writer` (in case it works) is an option?

Reply:
@seliopou It's been a little while, but IIRC one of the tests I added will break if this line is removed, so I'd say a different PR might be unnecessary.

Reply:
I don't know if `wakeup_writer` will work here, which is why the existing, correct behavior should be tested independently before this PR is worked on further.

Reply:
Ah, I understand. Thanks

Reply:
Since the behavior that broke still isn’t tested, that behavior should be tested in a separate PR before this one is reviewed any further.

Reply:
Suppose you changed nothing except for this line in the way you originally did. That would break existing behavior in the two ways I described. It may not be reachable, but it’s a big laying in wait.

So there should be a test for those cases independent of the other changes in this PR, and the it can be expanded to cover additional codepaths in this PR as necessary.

Reply:
That's really helpful, thanks for dumbing it down for me. I'll do that in a separate PR